packages/cli-hydrogen/src/cli/services/deploy/config.ts

```diff
@@ -0,0 +1,59 @@
+import {DeployConfig, ReqDeployConfig} from './types.js'
+import {git, path, error} from '@shopify/cli-kit'
+
+const simpleGit = git.factory()
+
+export const getDeployConfig = async (config: DeployConfig): Promise<ReqDeployConfig> => {
+  await git.ensurePresentOrAbort()
+  await git.ensureInsideGitDirectory()
+
+  const [latestCommit, repository, commitRef] = await Promise.all([getLatestCommit(), getRepository(), getHeadRef()])
+
+  const directory = config.path ? path.resolve(config.path) : process.cwd()
+
+  return {
+    deploymentToken: config.deploymentToken,
+    dmsAddress: config.dmsAddress,
+    commitMessage: config.commitMessage ?? latestCommit.message,
+    commitAuthor: config.commitAuthor ?? latestCommit.author_name,
+    commitSha: latestCommit.hash,
+    commitRef,
+    timestamp: latestCommit.date,
+    repository,
+    path: directory,
+  }
+}
+
+const getLatestCommit = async () => {
```

Comment:
All these internal utility functions are useful beyond the Hydrogen domains o I recommend moving it down to the `git` module inside `@shopify/cli-kit`.